### PR TITLE
Check-in: Get current location of user C#

### PR DESCRIPTION
**Pull request by NikitaJain-MSFT**
Use mobile SDK that allows IWs to perform a check-in – capturing current time, current location and submitting it as adaptive card
also get all last check in's of user.

### CODE REVIEW

File: samples/check-in-current-location/csharp/CheckInLocation/AppPackage/manifest.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
+﻿{
+  "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.10/MicrosoftTeams.schema.json",
+  "manifestVersion": "1.10",
+  "version": "1.0.0",
+  "id": "{{Microsoft-App-Id}}",
+  "packageName": "com.microsoft.teams.checkinlocation",
```
Comment (AbhijeetThakur-msft) [inline samples/check-in-current-location/csharp/CheckInLocation/AppPackage/manifest.json:6]: Let's make app name consistent here as well

File: samples/check-in-current-location/csharp/CheckInLocation/AppPackage/manifest.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      "isNotificationOnly": false
+    }
+  ],
+  "permissions": [ "identity", "messageTeamMembers" ],
+  "devicePermissions": [
+    "geolocation"
+  ],
+  "validDomains": [
+    "{{domain-name}}",
+    "token.botframework.com"
```
Comment (AbhijeetThakur-msft) [inline samples/check-in-current-location/csharp/CheckInLocation/AppPackage/manifest.json:41]: will we require token.... in valid domain?

File: samples/check-in-current-location/csharp/CheckInLocation/Bots/ActivityBot.cs [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+        /// </summary>
+        /// <param name="turnContext">The turn context.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
+        {
+            if(turnContext.Activity.Text.ToLower() == "view last check in")
+            {
+                var fileName = Path.Combine(_env.ContentRootPath, $".\\wwwroot\\UserLocationdetails.json");
+                string Json = File.ReadAllText(fileName);
```
Comment (AbhijeetThakur-msft) [inline samples/check-in-current-location/csharp/CheckInLocation/Bots/ActivityBot.cs:48]: Let's follow variable naming convention

File: samples/check-in-current-location/csharp/CheckInLocation/Bots/ActivityBot.cs [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+        /// Handle when a message is addressed to the bot.
+        /// </summary>
+        /// <param name="turnContext">The turn context.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
+        {
+            if(turnContext.Activity.Text.ToLower() == "view last check in")
+            {
+                var fileName = Path.Combine(_env.ContentRootPath, $".\\wwwroot\\UserLocationdetails.json");
```
Comment (AbhijeetThakur-msft) [inline samples/check-in-current-location/csharp/CheckInLocation/Bots/ActivityBot.cs:47]: let's rename this to checkindetails

File: samples/check-in-current-location/csharp/CheckInLocation/Bots/ActivityBot.cs [hunk 5]
```diff
@@ -0,0 +1,344 @@
...
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects or threads to receive notice of cancellation.</param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        /// <remarks>
+        protected override async Task OnMembersAddedAsync(IList<ChannelAccount> membersAdded, ITurnContext<IConversationUpdateActivity> turnContext, CancellationToken cancellationToken)
+        {
+            foreach (var member in turnContext.Activity.MembersAdded)
+            {
+                if (member.Id != turnContext.Activity.Recipient.Id)
+                {
+                    await turnContext.SendActivityAsync(MessageFactory.Text($"Hello and welcome! With this sample your bot can get your current location and all your last check in's. Please type 'hey' or 'check in' to get check in card or type 'view last check in' to get all check in details"), cancellationToken);
```
Comment (AbhijeetThakur-msft) [inline samples/check-in-current-location/csharp/CheckInLocation/Bots/ActivityBot.cs]: we can rephrase this to cover checkin and view checkin scenarios

File: samples/check-in-current-location/csharp/CheckInLocation/Bots/ActivityBot.cs [hunk 6]
```diff
@@ -0,0 +1,344 @@
...
+                UserName = user,
+                Longitude = longitude,
+                Latitude = latitude,
+                UserId = turnContext.Activity.From.AadObjectId
+            };
+
+            SaveUserDetails(userDetails);
+            await turnContext.SendActivityAsync(MessageFactory.Attachment(GetAdaptiveCardForUserLocation(time, user, latitude, longitude)), cancellationToken);
+
+            return null;
```
Comment (AbhijeetThakur-msft) [inline samples/check-in-current-location/csharp/CheckInLocation/Bots/ActivityBot.cs:176]: we can remove this null return here

File: samples/check-in-current-location/csharp/README.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
+﻿﻿﻿
+## Get Check-in info of user
+
+This sample demos to get current location and time of user and all last check in of user.
```
Comment (AbhijeetThakur-msft) [inline samples/check-in-current-location/csharp/README.md:4]: Let's rephrase this as well

File: samples/check-in-current-location/csharp/CheckInLocation/Bots/ActivityBot.cs [hunk 8]
```diff
@@ -0,0 +1,347 @@
...
+            }
+
+            return attachmentList;
+        }
+
+        // Save user details in json file.
+        private void SaveUserDetails(UserDetails userDetails)
+        {
+            var fileName = Path.Combine(_env.ContentRootPath, $".\\wwwroot\\UserLocationdetails.json");
+            string Json = File.ReadAllText(fileName);
```
Comment (v-chetsh) [inline samples/check-in-current-location/csharp/CheckInLocation/Bots/ActivityBot.cs:332]: Let's follow naming conventions and rename Json with proper name

File: samples/check-in-current-location/csharp/CheckInLocation/Models/AdaptiveCardAction.cs [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+namespace CheckInLocation.Models
+{
+    using Microsoft.Bot.Schema;
+    using Newtonsoft.Json;
+
+    /// <summary>
+    /// Adaptive card action model class.
+    /// </summary>
+    public class AdaptiveCardAction
```
Comment (v-chetsh) [inline samples/check-in-current-location/csharp/CheckInLocation/Models/AdaptiveCardAction.cs:13]: Why this class contains Longitude & Latitude, are these actions?

File: samples/check-in-current-location/csharp/CheckInLocation/Models/UserDetails.cs [hunk 10]
```diff
@@ -0,0 +1,22 @@
+﻿// <copyright file="UserDetails.cs" company="Microsoft">
+// Copyright (c) Microsoft. All rights reserved.
+// </copyright>
+
+namespace CheckInLocation.Models
+{
+    /// <summary>
+    /// User Details model class.
+    /// </summary>
+    public class UserDetails
```
Comment (v-chetsh) [inline samples/check-in-current-location/csharp/CheckInLocation/Models/UserDetails.cs:10]: It should be UserDetail

File: samples/check-in-current-location/csharp/CheckInLocation/appsettings.json [hunk 11]
```diff
@@ -0,0 +1,5 @@
+{
+  "MicrosoftAppId": "{{Microsoft-App-Id}}",
+  "MicrosoftAppPassword": "{{Microsoft-App-Password}}",
+  "ApplicationBaseUrl": ""
```
Comment (v-chetsh) [inline samples/check-in-current-location/csharp/CheckInLocation/appsettings.json:4]: Either add placeholder for ApplicationBaseUrl or remove it from others.

File: samples/check-in-current-location/csharp/README.md [hunk 12]
```diff
@@ -0,0 +1,122 @@
+﻿﻿﻿
+## Get Check-in info of user
```
Comment (v-chetsh) [inline samples/check-in-current-location/csharp/README.md:2]: Add proper naming conventions for "UserLocation**d**etails.json" file

File: samples/check-in-current-location/csharp/README.md [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+
+This sample demos to get current location and time of user and all last check in of user.
+
+`Currently, Microsoft Teams support for get geolocation capability is only available for mobile clients`
+
+![Check in card](CheckInLocation/Images/CheckinCard.png)
```
Comment (v-chetsh) [inline samples/check-in-current-location/csharp/README.md:8]: Please make sure if you have properly link  your readme with root file.

**Review by v-chetsh — CHANGES_REQUESTED** [review]
Reviewed & provided my comments.

File: samples/check-in-current-location/csharp/CheckInLocation/Models/AdaptiveCardAction.cs (hunk 9 above)
Comment (NikitaJain-MSFT) [reply samples/check-in-current-location/csharp/CheckInLocation/Models/AdaptiveCardAction.cs]: Longitude & Latitude needs to pass fetch/async method for view location hence added in this class

**Review by v-chetsh — APPROVED** [review]
Changes looks fine, approving it.

**Review by Wajeed-msft — APPROVED** [review]
